web_src/js/features/common-global.js

```diff
@@ -142,6 +142,18 @@ export function initGlobalCommon() {
       window.location = href;
     }
   });
+
+  // loading-button this logic used to prevent push one form more than one time
+  $(document).on('click', '.button.loading-button', function (e) {
+    const $btn = $(this);
+
+    if ($btn.hasClass('loading')) {
+      e.preventDefault();
+      return false;
+    }
```

Comment:
I think you need capture instead of bubble here. Otherwise the event hanlder on the button would be executed first (for example: ajax)

https://stackoverflow.com/questions/4616694/what-is-event-bubbling-and-capturing

Reply:
To be clear, what I mean is:

If we are trying to introduce a general method, should the all cases be considered together? Will this mechanism be used for ajax buttons in future?

* If this mechanism is for traditional forms (non-ajax) only, these CSS names are too broad, it will be mis-used by ajax buttons in future.

* If this mechanism is for both traditional forms and ajax requests, it could be fine tuned some more.

Reply:
I think it should be generic for all use cases where technically possible and needed

Reply:
Either is fine to me:

 * Make the CSS name in this PR clear for only traditional `form` buttons, and get this PR merged.
 * (OR) Let's spend some more time to fine tune some more.

How do you think?

Reply:
> Either is fine to me:
> 
> * Make the CSS name in this PR clear for only traditional `form` buttons, and get this PR merged.
> * (OR) Let's spend some more time to fine tune some more.
> 
> How do you think?

how about limit it by this way?
```suggestion
  // loading-button this logic used to prevent push one form more than one time
  $(document).on('click', 'form .button.loading-button', function (e) {
    const $btn = $(this);

    if ($btn.hasClass('loading')) {
      e.preventDefault();
      return false;
    }
```

Reply:
`loading-button` is not a proper name ......

If you are reading some code, when you see a button called `loading-button`, you might think it is a button which is loading something.  BUT, the purpose of the class is to make the button can only submit the form once.

I would suggest to use a clear name to reduce misunderstanding.